src/include/sof/alloc.h

```diff
@@ -130,57 +130,14 @@ void *_balloc(int zone, uint32_t caps, size_t bytes);
 void rfree(void *ptr);
 
 #if CONFIG_DEBUG_HEAP
-
-#define rmalloc(zone, caps, bytes)			\
-	({void *_ptr;					\
-	do {						\
-		_ptr = _malloc(zone, caps, bytes);	\
-		if (!_ptr) {				\
-			trace_error(TRACE_CLASS_MEM,	\
-				   "failed to alloc 0x%x bytes caps 0x%x", \
-				   bytes, caps);	\
-			alloc_trace_runtime_heap(zone, caps, bytes);	\
-		}					\
-	} while (0);					\
-	_ptr; })
-
-#define rzalloc(zone, caps, bytes)			\
-	({void *_ptr;					\
-	do {						\
-		_ptr = _zalloc(zone, caps, bytes);	\
-		if (!_ptr) {				\
-			trace_error(TRACE_CLASS_MEM,	\
-				   "failed to alloc 0x%x bytes caps 0x%x", \
-				   bytes, caps);	\
-			alloc_trace_runtime_heap(zone, caps, bytes);	\
-		}					\
-	} while (0);					\
-	_ptr; })
-
-#define rballoc(zone, caps, bytes)			\
-	({void *_ptr;					\
-	do {						\
-		_ptr = _balloc(zone, caps, bytes);	\
-		if (!_ptr) {				\
-			trace_error(TRACE_CLASS_MEM,	\
-				   "failed to alloc 0x%x bytes caps 0x%x", \
-				   bytes, caps);	\
-			alloc_trace_buffer_heap(zone, caps, bytes);	\
-		}					\
-	} while (0);					\
-	_ptr; })
-
 void alloc_trace_runtime_heap(int zone, uint32_t caps, size_t bytes);
 void alloc_trace_buffer_heap(int zone, uint32_t caps, size_t bytes);
-
-#else
-
-#define rmalloc(zone, caps, bytes)	_malloc(zone, caps, bytes)
-#define rzalloc(zone, caps, bytes)	_zalloc(zone, caps, bytes)
-#define rballoc(zone, caps, bytes)	_balloc(zone, caps, bytes)
```

Comment:
Why are you removing all the heap debug ?

Reply:
update, move the heap debug flag to rmalloc and rzalloc

Reply:
Nak, why  are you removing this ? The macro gives the line and file of the caller (making it easy to debug any failures).